src/controllers/controller.bar.js

```diff
@@ -223,15 +225,33 @@ module.exports = function(Chart) {
 				}
 			}
 
-			return stacks.length;
+			return stacks;
+		},
+
+		/**
+		 * Returns the effective number of stacks based on groups and bar visibility.
+		 * @private
+		 */
+		getStackCount: function() {
+			return this._getStacks().length;
 		},
 
 		/**
 		 * Returns the stack index for the given dataset based on groups and bar visibility.
+		 * @param {Number=} [datasetIndex] - The dataset index
```

Comment:
[nit] what the purpose of `=` in `{Number=}`?

Reply:
Indicates that it's optional: http://usejsdoc.org/tags-type.html

Though I don't think we need both the equals and square brackets. That would seem redundant to me. I would think we should just use one or the other. I was more familiar with the `=` syntax

Reply:
Didn't take time to look around, thinking it was a typo, thanks @benmccann! I agree, it's redundant, and I think we already adopted the other form `{Number} [datasetIndex] - ...`, so I would continue with that one.